File: DSA-DAILY/31-07-2025/898. Bitwise ORs of Subarrays @Leetcode-Med.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int subarrayBitwiseORs(vector<int>& arr) {
        unordered_set<int> res,curr;
        for(int i:arr){
            unordered_set<int>temp;
            temp.insert(i);
            for(int j:curr){
                temp.insert(i|j);
            }
            res.insert(temp.begin(),temp.end());
            curr=temp;
        }
        return res.size();
    }
};
int main() {
    Solution sol;
    int n;
    cout << "Enter size of array: ";
    cin >> n;
    cout << "Enter Array: ";
    vector<int> arr(n);
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    int result = sol.subarrayBitwiseORs(arr);
    cout << "Number of distinct bitwise ORs of subarrays: " << result << endl;
    return 0;
}

File: DSA-DAILY/31-07-2025/898. Bitwise ORs of Subarrays @Leetcode-Med.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var subarrayBitwiseORs = function(arr) {
    const res=new Set();
    const n = arr.length;

    for(let i=0;i<n;i++){
        res.add(arr[i]);
        let curr=arr[i];
        for(let j=i-1; j>=0; j--){
            if((arr[j]|curr) === arr[j])break;
            arr[j] |= curr;
            res.add(arr[j]);
        }
    }
    return res.size
};

const readInput = () => {
    const input = prompt("Enter an array of numbers separated by commas:");
    const arr = input.split(",").map(Number);
    return arr;
};

const main = () => {
    const arr = readInput();
    const result = subarrayBitwiseORs(arr);
    console.log("Number of distinct bitwise ORs:", result);
};
